tools/buildFactory.ts

```diff
@@ -174,73 +202,166 @@ export async function gulpUmdMinFactory(input: string, output: string) {
 }
 
 export function gulpIsolateFactory(
-  taskPrefix: string,
+  format: 'cjs' | 'esm',
   input: {
-    sourceRoots: string | string[],
     indexFile: string,
+    isolateBuildConfig: {
+      builds: {
+        name: string,
+        sourceRoot: string
+      }[]
+    },
+    buildName: string,
     globs: string | string[],
     opts?: SrcOptions
   },
   output: string,
-  overrideSettings?: Omit<ts.Settings, 'outDir'>,
-  useCjsTransform = false
+  overrideSettings?: Omit<gulpTs.Settings, 'outDir'>
 ) {
-  const dependencySolver = new DependencySolver(
-    input.sourceRoots,
-    {
-      baseUrl: input.opts?.base ?? '.',
-      outDir: output
-    }
-  );
+  output = `dist/isolate/${output}`;
+  const indexFileDir = 'dist/isolate/.build-data';
 
-  function build() {
-    return gulpFactory(
-      input,
-      output,
-      {
-        ...overrideSettings,
-        getCustomTransformers: (program?: Program) => {
-          const treeShakerTransform = tsTreeShaker({
-            solver: dependencySolver,
-            indexFile: input.indexFile
-          });
-
-          const customTransformers = overrideSettings?.getCustomTransformers?.(program);
-          return {
-            before: customTransformers?.before,
-            after: [
-              ...treeShakerTransform.after,
-              ...customTransformers?.after ?? []
-            ],
-            afterDeclarations: [
-              ...treeShakerTransform.afterDeclarations,
-              ...customTransformers?.afterDeclarations ?? []
-            ]
-          };
-        }
-      },
-      useCjsTransform
-    );
+  async function generateIndex() {
+    // get files from glob
+    const globs = Array.isArray(input.globs) ? input.globs : [input.globs];
+    const files = globs
+      .map((pattern) => glob.sync(pattern, input.opts))
+      .reduce((acc, val) => acc.concat(val), []);
+    // generate sha256 hash of the source code
+    const hash = crypto.createHash('sha256');
```

Comment:
What's the purpose of the hash?

Reply:
Creating a treeshaked index file takes a long time, so I process it all at once.

In future builds, you can skip the build by comparing the hash values.

Reply:
It's just use for local build, does it?